fields/types/gcsimage/GcsImageType.js

```diff
@@ -66,405 +66,336 @@ Object.defineProperty(gcsimage.prototype, 'gcsConfig', {
-							}
-							return callback(null, imageData);
-						});
-			} else {
-				if (update) {
-					item.set(_this.path, imageData);
-				}
-				callback(null, imageData);
-			}
-		}).catch(function (err) {
-			console.error('UPLOADING ERROR:', err);
-			bucket.deleteFiles({
-				prefix: gcsDir + filenameWithoutExt,
-			}, function (deleteErr) {
-				if (deleteErr) {
-					return callback(deleteErr);
-				}
-				callback(err);
-			});
-		}).then(function () {
-			// delete local file
-			console.log('DELETE LOCAL FILE:', file.path);
-			try {
-				fs.unlinkSync(file.path);
-			} catch (err) {
-				console.error('DELETE LOCAL FILE ERROR:', err);
-			};
-		});
-	};
-
-	doUpload();
+  var _this = this;
+  var ONE_YEAR = 60 * 60 * 24 * 365;
+  var gcsDir = this.options.destination ? this.options.destination : '';
+  var isPublicRead = _this.options.publicRead ? _this.options.publicRead : false;
+  var prefix = _this.options.datePrefix ? moment().format(_this.options.datePrefix) + '-' : '';
+  var filename = prefix + file.name;
+  var split = filename.split('.');
+  var filenameWithoutExt = split[0];
+  var ext = split[1] || '';
+  var originalname = file.originalname;
+  var filetype = file.mimetype || file.type;
+
+  if (typeof update === 'function') {
+    callback = update;
+    update = false;
+  }
+
+  if ((_this.options.allowedTypes && !_.indexOf(_this.options.allowedTypes, filetype)
+    || filetype.match('image.*') === null)) {
+    return callback(new Error('Unsupported File Type: ' + filetype));
+  }
+
+  var doUpload = function () {
+
+    if (typeof _this.options.filename === 'function') {
+      filename = _this.options.filename(item, filename, originalname);
+    }
+
+    var bucket = gcsHelper.initBucket(_this.gcsConfig, _this.options.bucket);
+    return gcsHelper.uploadFileToBucket(bucket, fs.createReadStream(file.path), {
+      destination: gcsDir + filename,
+      filetype: filetype,
+      isPublicRead: isPublicRead,
+      cacheControl: 'public, max-age=' + ONE_YEAR,
+    }).then(function () {
+      const dimensions = sizeOf.sync(fs.readFileSync(file.path));
+
+      var imageData = {
+        filepath: file.path,
+        filename: filename,
+        filetype: filetype,
+        height: dimensions.height,
+        size: file.size,
+        url: gcsHelper.getPublicUrl(_this.options.bucket, gcsDir + filename),
+        width: dimensions.width,
+      };
+
+      if (_this.options.hooks && typeof _this.options.hooks.postUpload === 'function') {
+        return _this.options.hooks.postUpload(imageData)
+          .then((postUploadData) => {
+            if (update) {
+              item.set(_this.path, postUploadData);
+            }
+            return callback(null, postUploadData);
+          })
+      } else {
+        if (update) {
+          item.set(_this.path, imageData);
+        }
+        callback(null, imageData);
+      }
+    }).catch(function (err) {
+      console.error('UPLOADING ERROR:', err);
```

Comment:
minor. Remove the extra annotating here since we already wraps the error information from plate.